scala-debugger-api/src/main/scala/org/scaladebugger/api/profiles/scala210/info/Scala210ObjectInfo.scala

```diff
@@ -40,54 +40,13 @@ class Scala210ObjectInfo(
    */
   override def isJavaInfo: Boolean = false
 
-
   /**
    * Returns all visible fields contained in this object.
    *
    * @note Provides no offset index information!
    * @return The profiles wrapping the visible fields in this object
    */
-  override def fields: Seq[FieldVariableInfo] = {
-    val baseFields = super.fields
-
-    // Will retrieve fields of these objects
-    val expandNames = Seq("$outer")
-
-    // Will skip these fields
-    // TODO: Provide better means of skipping executionStart
-    //       since this avoids it even if someone added it directly
-    val ignoreNames = Seq("MODULE$", "executionStart", "serialVersionUID")
-
-    // Will skip fields whose names start with these prefixes
-    val ignoreNamePrefix = Seq("scala$")
-
-    // Will skip fields whose origin starts with these strings
-    val ignoreOrigin = Seq("scala.")
-
-    // TODO: Handle infinitely-recursive fields when expanding
-    def transformField(field: FieldVariableInfo): Seq[FieldVariableInfo] = {
-      val jField = field.toJavaInfo
-      val value = field.toValueInfo
-
-      // If the field is something we should ignore directly, do so
-      if (ignoreNames.contains(jField.name)) Nil
-
-      // If the field name starts with a prefix we should ignore, do so
-      else if (ignoreNamePrefix.exists(jField.name.startsWith)) Nil
-
-      // If the field's origin starts with an origin we don't want, ignore it
-      else if (ignoreOrigin.exists(jField.declaringType.name.startsWith)) Nil
-
-      // If the field is one that should be expanded, do so
-      else if (value.isObject && expandNames.contains(jField.name))
-        value.toObjectInfo.fields.flatMap(transformField)
-
-      // Otherwise, return the normal field
-      else Seq(field)
-    }
-
-    baseFields.flatMap(transformField)
-  }
+  override def fields: Seq[FieldVariableInfo] = super.fields.flatMap(transformField(_))
```

Comment:
I know that I had `super.fields` used earlier, but what about using `toJavaInfo.fields` instead of `super.fields`? I added that method to provide a guaranteed way to get back to the raw Java implementation. This way, if Scala210 (or other profiles) don't directly inherit from the _Pure_ profile, this logic won't be affected. Just popped in my head when looking at the code changes.